libafl/src/executors/forkserver.rs

```diff
@@ -0,0 +1,464 @@
+
+        let out_file = OutFile::new(&out_filename);
+
+        //forkserver
+        let mut forkserver = Forkserver::new(
+            target.clone(),
+            args.clone(),
+            out_file.as_raw_fd(),
+            use_stdin,
+            0,
+        );
+        let (rlen, _) = forkserver.read_st()?; //initial handshake
+
+        if rlen == 4 {
+            println!("Forkserver up!!");
+        }
+
+        Ok(Self {
+            target,
+            args,
+            out_file,
+            forkserver,
+            observers,
+            phantom: PhantomData,
+        })
+    }
+
+    pub fn target(&self) -> &String {
+        &self.target
+    }
+
+    pub fn args(&self) -> &Vec<String> {
+        &self.args
+    }
+
+    pub fn forkserver(&self) -> &Forkserver {
+        &self.forkserver
+    }
+}
+
+impl<I, OC, OF, OT, S> Executor<I> for ForkserverExecutor<I, OC, OF, OT, S>
+where
+    I: Input + HasTargetBytes,
+    OC: Corpus<I>,
+    OF: Feedback<I, S>,
+    OT: ObserversTuple,
+{
+    #[inline]
+    fn run_target(&mut self, _input: &I) -> Result<ExitKind, Error> {
+        let t1 = current_time();
+        let slen = self
+            .forkserver
+            .write_ctl(self.forkserver().last_run_timed_out())?;
+        if slen != 4 {
+            panic!("failed to request new process");
+        }
+
+        let (rlen, pid) = self.forkserver.read_st()?;
+        if rlen != 4 {
+            panic!("failed to request new process")
+        }
+
+        if pid <= 0 {
+            panic!("forkserver is misbehaving");
+        }
+        //println!("pid: {:#?}",pid);
+
+        //child running
+
+        let (_, status) = self.forkserver.read_st()?;
+        self.forkserver.status = status;
+
+        let t2 = current_time();
+        println!("Exec time {} ns", (t2 - t1).as_nanos());
+
+        Ok(ExitKind::Ok)
+    }
+}
+
+impl<EM, I, OC, OF, OT, S, Z> HasExecHooks<EM, I, S, Z> for ForkserverExecutor<I, OC, OF, OT, S>
+where
+    EM: EventFirer<I, S> + EventRestarter<S>,
+    I: Input + HasTargetBytes,
+    OC: Corpus<I>,
+    OF: Feedback<I, S>,
+    OT: ObserversTuple,
+    S: HasSolutions<OC, I>,
+    Z: HasObjective<I, OF, S>,
+{
+    #[inline]
+    fn pre_exec(
+        &mut self,
+        _fuzzer: &mut Z,
+        _state: &mut S,
+        _event_mgr: &mut EM,
+        input: &I,
+    ) -> Result<(), Error> {
+        //write to test case
+        self.out_file
+            .write_buf(&input.target_bytes().as_slice().to_vec());
+        //outfile gets automatically closed.
+        Ok(())
+    }
+
+    fn post_exec(
+        &mut self,
+        fuzzer: &mut Z,
+        state: &mut S,
+        event_mgr: &mut EM,
+        input: &I,
+    ) -> Result<(), Error> {
+        if !libc::WIFSTOPPED(self.forkserver.status()) {
+            self.forkserver.child_pid = 0;
+        }
+
+        if libc::WIFSIGNALED(self.forkserver.status()) {
+            let interesting = fuzzer
```

Comment:
This is not needed, Fuzzer already calls is_interesting for the objective https://github.com/AFLplusplus/LibAFL/blob/main/libafl/src/fuzzer.rs#L559